### PR TITLE
Hibernate

### PR DESCRIPTION
**Pull request by VolhaMan**
@manusovich 

Database layer has been migrated to hibernate

### CODE REVIEW

**Comment by manusovich** [comment]
Remove from PR files which should not be in repository such as .DS_Store or idea xml

File: pom.xml [hunk 1]
```diff
@@ -7,6 +7,11 @@
     <version>1.0-SNAPSHOT</version>
     <name>weather Maven Webapp</name>
     <url>http://maven.apache.org</url>
+    <properties>
+        <spring.security.version>3.2.3.RELEASE</spring.security.version>
```
Comment (manusovich) [inline pom.xml]: be consistent. if you defining versions here - define them for everything

File: src/main/java/weather/ManageWeather.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        Transaction transaction = null;
+        List<CurrentWeather> history = null;
+
+        try {
+            transaction = session.beginTransaction();
+            history = new LinkedList<>(session.createQuery("FROM weather.CurrentWeather").list());
+            transaction.commit();
+        } catch (HibernateException ex) {
+            if (transaction != null) {
+                transaction.rollback();
```
Comment (manusovich) [inline src/main/java/weather/ManageWeather.java]: rollback for what? looks like you are misunderstanding what is transaction menas

File: src/main/webapp/WEB-INF/pages/login.jsp [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<html>
+<head>
+    <title>Login Page</title>
+    <style>
+        .error {
```
Comment (manusovich) [inline src/main/webapp/WEB-INF/pages/login.jsp]: use separate css file

File: src/main/webapp/WEB-INF/pages/login.jsp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            background: #fff;
+            -webkit-border-radius: 2px;
+            -moz-border-radius: 2px;
+            border: 1px solid #000;
+        }
+    </style>
+</head>
+<body onload='document.loginForm.username.focus();'>
+
+<h1>Spring Security Custom Login Form (XML)</h1>
```
Comment (manusovich) [inline src/main/webapp/WEB-INF/pages/login.jsp]: why is it XML form? why you called that XML?

File: src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<hibernate-configuration>
+    <session-factory>
+        <property name="hibernate.dialect">
+            org.hibernate.dialect.PostgreSQLDialect
+        </property>
+        <property name="hibernate.connection.driver_class">
+            org.postgresql.Driver
+        </property>
+
+        <!-- List of XML mapping files -->
```
Comment (manusovich) [inline src/main/resources/hibernate.cfg.xml]: where is XML mapping files?

File: src/main/webapp/WEB-INF/dispatcher-servlet.xml [hunk 6]
```diff
@@ -25,4 +25,12 @@
             <value>classpath:app.properties</value>
         </property>
     </bean>
+    <bean class="org.springframework.web.servlet.view.InternalResourceViewResolver">
+        <property name="prefix">
+            <value>/WEB-INF/pages/</value>
```
Comment (manusovich) [inline src/main/webapp/WEB-INF/dispatcher-servlet.xml]: NEVER put jsp to the WEB-INF folder. Don't put any resources there

**Comment by VolhaMan** [comment]
@manusovich 
Project has been updated according your comments

**Pull request merged**